server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -139,7 +139,7 @@ public void writeAtomic(T obj, BlobContainer blobContainer, String name) throws
         } catch (IOException ex) {
             // Move failed - try cleaning up
             try {
-                blobContainer.deleteBlob(tempBlobName);
+                blobContainer.deleteBlobIgnoringIfNotExists(tempBlobName);
```

Comment:
this does not change anything here? We are already catching the NoSuchFileException in the line below, which is an IOException.

Reply:
Right - in this case it must not swallow the NoSuchFileException